### PR TITLE
Add CloudWatch 101 guide

### PR DESCRIPTION
**Pull request by zachmargolis**
Adds a from-scratch guide to how to query CloudWatch logs.

This is a first pass, I'm sure it's missing something, we can iterate and add things as we discover them

[preview of page](https://federalist-7d0b2b76-42fc-4df1-9825-8c3cd77e0a15.sites.pages.cloud.gov/preview/gsa-tts/identity-handbook/margolis-cloudwatch-101/articles/cloudwatch-101.html)

### CODE REVIEW

File: _articles/appdev-cloudwatch-dashboard-guide.md [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
 
-Amazon Cloudwatch makes it possible to search Login.gov logs and AWS metrics to
+Amazon CloudWatch makes it possible to search Login.gov logs and AWS metrics to
 create visualizations or display statistics.
-Cloudwatch Insights is the tool that makes it possible to query logs.
-Information on how to build Cloudwatch Insights queries can be found in the
-[AWS documentation](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/CWL_QuerySyntax.html).
+CloudWatch Insights is the tool that makes it possible to query logs. See [CloudWatch 101][cloudwatch-101] for more information on how to write queries.
+
+[cloudwatch-101]: {% link _articles/cloudwatch-101.md %}
```
Comment (mww59) [inline _articles/appdev-cloudwatch-dashboard-guide.md:15]: We have a page in the `identity-devops` wiki that covers [some of this](https://gitlab.login.gov/lg/identity-devops/-/wikis/Guide:-Cloudwatch-Logs,-Metrics-and-Dashboards), it appears that it's not linked anywhere in the handbook.  Should we include it here as well?

Comment (zachmargolis) [reply _articles/appdev-cloudwatch-dashboard-guide.md:15]: ooo I was not aware of that page! thanks for the link, I added a "See Also" in 3c99e540fc293c6b38f8bd1b7ce414596f6c6946

Comment (zachmargolis) [reply _articles/appdev-cloudwatch-dashboard-guide.md:15]: oh I got confused about which page, I added a link from this dashboards article here in c5cacd8f1a4e9730b743bc10de53159c94156742

File: _articles/cloudwatch-101.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+    ![screenshot showing the Logs Insights menu item in CloudWatch]({{ site.baseurl }}/images/aws-cloudwatch-logs-insights.png)
+
+### Running a query
+
+![screenshot of the query interface for CloudWatch Insights]({{ site.baseurl }}/images/aws-cloudwatch-query.png)
+
+1. Make sure to select a log group. For most queries, we want **prod_/srv/idp/shared/log/events.log**. In the "Select up to 50 log groups" combobox, type in "events.log" to filter down the list and select the **prod_** one.
+
+1. Set the time range. For consistency across timezones, we recommend the **UTC** timezone.
```
Comment (benjaminchait) [inline _articles/cloudwatch-101.md:41]: How do you feel about suggesting "absolute" date ranges (rather than "relative" date ranges)?

Comment (zachmargolis) [reply _articles/cloudwatch-101.md:41]: I think for a lot a quick exploratory cases, relative date range are easy. For queries that we want to refer back to (incidents, specific A/B test runs) absolute makes more sense

File: _articles/cloudwatch-101.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+#### `count_distinct`
+
+Briefly: **Do not use `count_distinct`, it is inaccurate**
+
+From the [AWS documentation on `count_distinct`][count-distinct]:
+
+> Returns the number of unique values for the field. If the field has very high cardinality (contains many unique values), the value returned by count_distinct is just an approximation.
+
+To work around this, consider doing post processing externally. Another option, if there are fewer than 10,000 things is to do:
```
Comment (aduth) [inline _articles/cloudwatch-101.md:138]: We  could also point to the `query-cloudwatch` script in IdP, which has an accurate version of [`--count-distinct`](https://github.com/18F/identity-idp/blob/9394620eeee7a3f6a6d03b4f475a329d83c1f499/bin/query-cloudwatch#L249-L251)?

File: _articles/cloudwatch-101.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+CloudWatch Insights does not support joins
+
+#### Max 10,000 rows
+
+[query-cloudwatch][query-cloudwatch] script with the `--complete` option
+
+#### Too many fields
+
+CloudwatchInsights gives up on parsing after there are too many fields in a blob. To work around it, export the full `@message` field and parse it externally, such as in Ruby
```
Comment (aduth) [inline _articles/cloudwatch-101.md:122]: Nit:

```suggestion
CloudWatch Insights gives up on parsing after there are too many fields in a blob. To work around it, export the full `@message` field and parse it externally, such as in Ruby
```

**Review by aduth — APPROVED** [review]
Looks great! I can see this being a handy reference for people new to CloudWatch

File: _articles/cloudwatch-101.md [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+
+### Getting to CloudWatch Insights
+
+1. Log in to AWS
+
+    ```bash
+    # depending on the roles you have pick between
+    aws-vault login prod-analytics
+    # or
+    aws-vault login prod-power
```
Comment (Sgtpluck) [inline _articles/cloudwatch-101.md:24]: maybe mention that if you log in this way (as opposed to through the console) you can't switch roles, if you need to go back and forth from prod to int?

**Review by Sgtpluck — APPROVED** [review]
this looks great! thank you so much!

File: _articles/cloudwatch-101.md (hunk 5 above)
Comment (zachmargolis) [reply _articles/cloudwatch-101.md:24]: idk how those work! this is literally the only way to log in that I know!

File: _articles/cloudwatch-101.md (hunk 3 above)
Comment (zachmargolis) [reply _articles/cloudwatch-101.md]: updated in 47941653d6e127fb94a438d31f3a98b4ae2c49ea

**Pull request merged**